js/src/viewer.js

```diff
@@ -77,43 +78,41 @@
       }
 
       // add workspace configuration
-      this.layout = typeof this.layout !== 'string' ? JSON.stringify(this.layout) : this.layout;
+      this.layout = typeof this.state.getStateProperty('layout') !== 'string' ? JSON.stringify(this.state.getStateProperty('layout')) : this.state.getStateProperty('layout');
       this.workspace = new $.Workspace({
         layoutDescription: this.layout.charAt(0) === '{' ? JSON.parse(this.layout) : $.layoutDescriptionFromGridString(this.layout), 
-        parent: this, 
-        appendTo: this.element.find('.mirador-viewer')
+        appendTo: this.element.find('.mirador-viewer'),
+        state: this.state
```

Comment:
Now we pass the state on to Workspace, which can access any information it needs, and pass the state on to Slot.